File: CsharpProjects/StateSolucao03/StateSolucao03/domains/state/EnviadoState.cs
using StateSolucao03.domains;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StateSolucao03.domains.state
{
    public class EnviadoState : State
    {
        private Pedido pedido;
        public EnviadoState(Pedido pedido)
        {
            this.pedido = pedido;
        }
        void State.cancelarPedido()
        {
            throw new Exception("Operacao não suportada, pedido já foi enviado");
        }
        void State.despacharPedido()
        {
            throw new Exception("Operacao não suportada, pedido já foi enviado");
        }
        void State.sucessoAoPagar()
        {
            throw new Exception("Operacao não suportada, pedido já foi enviado");
        }
    }
}

File: JavaProjects/StateSolucao03/src/domains/state/EnviadoState.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package domains.state;

import domains.Pedido;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author jeffe
 */
public class EnviadoState implements State{

    private Pedido pedido;

    public EnviadoState(Pedido pedido) {
        this.pedido = pedido;
    }
    
    @Override
    public void sucessoAoPagar() {
        try {
            throw new Exception("Operação não suportada - pedido cancelado");
        } catch (Exception ex) {
            Logger.getLogger(CanceladoState.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    @Override
    public void cancelarPedido() {
        try {
            throw new Exception("Operação não suportada - pedido cancelado");
        } catch (Exception ex) {
            Logger.getLogger(CanceladoState.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    @Override
    public void despacharPedido() {
        try {
            throw new Exception("Operação não suportada - pedido cancelado");
        } catch (Exception ex) {
            Logger.getLogger(CanceladoState.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
}
